### PR TITLE
feat: added ccra schema to global-api

### PR DESCRIPTION
**Pull request by amanda-eames**
- Need to review with the team before merging

### CODE REVIEW

File: global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          latest_year INT,
+          scenario_name VARCHAR,
+          CONSTRAINT unique_impactchain UNIQUE (id)
+        );
+    """))
+
+    conn.execute(text("""
+    CREATE TABLE IF NOT EXISTS modelled.ccra_impactchain_indicator (
+      impact_id UUID,
+      indicator_id UUID,
```
Comment (lemilonkh) [inline global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py:54]: Which one is the primary key in this table? Or is it a compound key?

File: global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      relationship VARCHAR,
+      datasource VARCHAR,
+      CONSTRAINT unique_impactchain_indicator UNIQUE (impact_id, indicator_id, category, subcategory)
+    );
+    """))
+
+    conn.execute(text("""
+    CREATE TABLE IF NOT EXISTS modelled.ccra_riskassessment (
+      "impact_id" uuid,
+      "actor_id" varchar,
```
Comment (lemilonkh) [inline global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py:70]: Does this table need its own private key or is the compound key (not fully specified in SQL though) sufficient?
This way there will be no PK index, only a unique index on the table I think.

File: global-api/routes/ccra_assessment.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    records = db_ccra_cities(country_code)
+
+    if not records:
+        raise HTTPException(status_code=404, detail="No data available")
+
+    # Transform database records into the desired format for the response
+    response = [
+            {
+                "cityName": record['city_name'],  # Assuming actor_id corresponds to city name
```
Comment (lemilonkh) [inline global-api/routes/ccra_assessment.py:118]: Can we use consistently either camelCase or snake_case for the keys and not mix it?

File: global-api/routes/ccra_assessment.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+    records = db_risk_assessment(actor_id, scenario_name)
+
+    if not records:
+        raise HTTPException(status_code=404, detail="No data available")
+
+    # Transform database records into the desired format for the response
+    response = [
+            {
+                "keyimpact": record['keyimpact_name'],  # Assuming actor_id corresponds to city name
```
Comment (lemilonkh) [inline global-api/routes/ccra_assessment.py:151]: This comment might be unrelated?

File: global-api/tests/test_ccra_assessment.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+# Mock the database functions
+@pytest.fixture(autouse=True)
+def mock_db_functions(monkeypatch):
+    monkeypatch.setattr("routes.ccra_assessment.db_risk_assessment", lambda actor_id, scenario_name: mock_risk_assessment_data)
+    monkeypatch.setattr("routes.ccra_assessment.db_impactchain_indicator", lambda actor_id, scenario_name: mock_impactchain_indicator_data)
+    monkeypatch.setattr("routes.ccra_assessment.db_ccra_cities", lambda country_code: mock_ccra_cities_data)
+
+# Test the /ccra/city/{country_code} endpoint
+def test_get_ccra_cities():
```
Comment (lemilonkh) [inline global-api/tests/test_ccra_assessment.py:65]: Great to have these tests, let's start executing them in CI next (in a future PR) :sparkles:

File: global-api/tests/test_main.py [hunk 6]
```diff
@@ -18,7 +18,7 @@ def test_health_check():
...
 # Test the data catalogue end point has no data
-def test_catalgue_no_data_available():
-    response = client.get("/api/v0/catalogue")
-    assert response.status_code == 404
-    assert response.json() == {"detail": "No data available"}
\ No newline at end of file
+# def test_catalgue_no_data_available():
+#     response = client.get("/api/v0/catalogue")
+#     assert response.status_code == 404
+#     assert response.json() == {"detail": "No data available"}
```
Comment (lemilonkh) [inline global-api/tests/test_main.py:24]: Should this be commented out or is it still to do?

**Review by lemilonkh — APPROVED** [review]
Great, thanks :sparkler: Thanks for adding all of the new documentation :+1:

File: global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py (hunk 1 above)
Comment (amanda-eames) [reply global-api/migrations/versions/b1e6ab175504_ccra_init_schema.py:54]: yeah it is a compounding key impact_id, indicator_id, category, subcategory

**Pull request merged**